proposals/nnnn-streaming-media-manager-update.md

```diff
@@ -0,0 +1,148 @@
+
+
+## Proposed solution
+#### Current Supported Features:
+- Starting/Stopping video session with different encryption flags
+- Starting/Stopping audio session with different encryption flags
+- Sending video data via `CVImageBufferRef`
+- Sending audio data via `NSData` of PCM data
+- Ability to interact with `SDLTouchManager`
+- Ability to set custom video encoder settings
+- Access to the shared `CVPixelBufferPoolRef` provided by the encoder
+- Access to the screen size of the connected head unit.
+
+#### New Features
+- Management of the video session lifecycle in conjunction with proxy & app state changes
+- Management of the audio session lifecycle in conjunction with proxy & app state changes
+- Ability to register to a touch manager handler for touch events.
+
+## Detailed design
+There are multiple parts of the proxy lifecycle that the SMM would have an impact on.
+#### `SDLLifecycleManager`
+If the application has an `SDLLifecycleConfiguration` with the `appType` of `NAVIGATION`, then SMM will be started automatically once `SDLLifecycleManager` enters the state `didEnterStateSettingUpManagers`. This would simply start the manager, not the sessions. At this point, SMM is ready and listening to HMI status changes and will internally start/stop protocols.
+#### `SDLStreamingMediaManager`
+Due to the need to make sure we properly unit test all possible combinations of HMI, app and streaming states, we need to be able to change the state machine at will. This, however, should not be exposed to the developer. Because of this, `SDLStreamingMediaManager` will be a wrapper around `SDLStreamingMediaLifecycleManager`, similar to how `SDLManager` is a wrapper around `SDLLifecycleManager`.
+
+#### `SDLStreamingMediaLifecycleManager`
+##### State Machines
+There will need to be multiple state machines that need to work together to deal with all the possible scenarios the app can be in on both the phone and head unit. Luckily, HMI states are already provided so we will not be building a new state machine, but will list the possible scenarios for different states below.
+###### Phone (new state machine)
+1. Resigning Active
+	- The app is moving to the background. If the streaming state is **Ready**, and in HMI **Limited or greater**, we should send ~30 frames so that we can alert the user to reopen the app. The streams will remain open. Can move to **Backgrounded**.
```

Comment:
30 frames guarantees that we send enough to be visible on the head unit. This can be changed, but in testing this seems to be a good amount.